src/cdk-lib/bedrock/agent.ts

```diff
@@ -587,23 +488,32 @@ export class Agent extends Construct implements cdk.ITaggableV2 {
   }
 
   /**
-   * Add an action group to the agent.
+   * Add knowledge base to the agent.
    */
-  public addActionGroup(props: AddAgentActionGroupProps): AgentActionGroup {
-    const actionGroupName = props.actionGroupName ?? generatePhysicalNameV2(
-      this,
-      'action-group',
-      { maxLength: 100, separator: '-' });
-    return new AgentActionGroup(this, `AgentActionGroup-${actionGroupName}`, {
-      agent: this,
-      actionGroupName,
-      description: props.description,
-      parentActionGroupSignature: props.parentActionGroupSignature,
-      actionGroupExecutor: props.actionGroupExecutor,
-      actionGroupState: props.actionGroupState,
-      apiSchema: props.apiSchema,
-      shouldPrepareAgent: this.shouldPrepareAgent,
+  public addKnowledgeBase(kb: KnowledgeBase) {
+    if (!kb.instruction) {
+      throw new Error('Agent Knowledge Bases require instructions.');
+    }
+    const agentKnowledgeBaseProperty: bedrock.CfnAgent.AgentKnowledgeBaseProperty = {
+      description: kb.description,
+      knowledgeBaseId: kb.knowledgeBaseId,
+      knowledgeBaseState: kb.knowledgeBaseState,
+    };
+
+    this.agent.knowledgeBases= [agentKnowledgeBaseProperty];
+  }
+
+
+  /**
+   * Add action group  to the agent.
+   */
+  public addActionGroups(props: AgentActionGroup) {
```

Comment:
Props should be an array of AgentActionGroup here

Reply:
done - 34db08e7b994c0a4d926f61a9cc9a395a3deba3e